elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexWorkVisitor.java

```diff
@@ -66,6 +68,25 @@
  */
 class ElasticsearchIndexWorkVisitor implements IndexWorkVisitor<Void, BackendRequest<?>> {
 
+	/*
+	 * Nesting marker field name: fields with this name have the parent path as their value,
+	 * and are used for denoting the parent element of the subsequent fields
+	 *
+	 * Use a very specific String to avoid naming conflict with user-defined field names
+	 */
+	public static final String NESTING_MARKER = "__Hibernate_nesting_marker";
+
+	/*
+	 * "this" marker field name: fields with this name are used for denoting that the
+	 * subsequent field represents an embedded as a whole, as opposed to simply a field
+	 * in an embedded.
+	 * The marker is necessary because in some cases, the embedded and its elements
+	 * share a common field name. For instance it is the case of embedded collections.
+	 *
+	 * Use a very specific String to avoid naming conflict with user-defined field names
+	 */
+	public static final String THIS_MARKER = "__Hibernate_this_marker";
```

Comment:
ObjectValue it is, then. I changed that and force-pushed.

Yes, the marker is dropped as part of HSEARCH-2397/#1188 (we don't need it anymore because the nesting marker carries more information).